theano/sandbox/cuda/dnn.py

```diff
@@ -443,8 +443,8 @@ def grad(self, inp, grads):
 
         top = gpu_contiguous(top)
 
-        d_img = GpuDnnConvGradI()(kerns, top, img.zeros_like(), desc)
-        d_kerns = GpuDnnConvGradW()(img, top, kerns.zeros_like(), desc)
+        d_img = GpuDnnConvGradI()(kerns, top, img, desc)
+        d_kerns = GpuDnnConvGradW()(img, top, kerns, desc)
```

Comment:
It might run one way or the other because the size of the variables is not taken into account for execution ordering, as far as I know.

I'm not sure there is a reasonable way to actually reuse the img buffer if there are no other replacements for the alloc_empty().